openehr-rm/src/main/java/com/nedap/archie/rm/support/identification/ObjectVersionId.java

```diff
@@ -17,4 +21,53 @@ public ObjectVersionId() {
     public ObjectVersionId(String value) {
         super(value);
     }
+
```

Comment:
Hmm, it may be nice to add 
```suggestion

public ObjectVersionId(String objectId, String creatingSystemId, String versionTreeId) {
        this(objectId + "::" + creatingSystemId + "::" + versionTreeId);
    }
    
```

here as well? :)

it also nicely self-documents the code a bit.

Reply:
That's a nice addition! I'll add it.

I was thinking about changing the constructor, too. We could avoid object creation with invalid values via maybe a static factory method. But that doesn't seem to align with Archie's API architecture. And I didn't want to break the API nevertheless.

Reply:
Yes, Archies API explicitly allows invalid objects. Very useful when parsing input from clients into models - even if it's invalid, you get a model that you can render to show errors. Plus EHR content can be modified and saved as a new version, so we decided the library needs to offer that capability.

I think in Archie we  just have to tell people that running the validations is very important and should not be skipped.

So I don't think static factory methods that do not allow for invalid values would be a good idea in these classes. Of course it is possible to do this in an external library, or even a different submodule of Archie. But that is quite a lot of work.



Probably good to add the constructor btw, not to change it, for backwards compatibility - and you probably know this - but always at least keep the no-argument constructor since that is necessary for XML and JSON parsing.

Reply:
Great, that makes sense. Thanks for the explanation!